CPLYTM-383 feat(plugin): adds client and server for policy plugin

gvauter opened the pull request:
Adds client and server for policy plugin and required transformations to/from the protobufs.

gvauter committed:
feat: adds client and server for policy plugin

Signed-off-by: George Vauter <gvauter@redhat.com>

jpower432 commented on plugin/transform_test.go:
```diff
@@ -0,0 +1,81 @@
+	}
+}
+
+func TestPolicyToProto(t *testing.T) {
```
I think we would need similar tests for the Results transformation.

jpower432 commented on plugin/transform_test.go:
```diff
@@ -0,0 +1,81 @@
+)
+
+// test harness for a plugin policy
+func loadPolicyPluginType() policy.Policy {
```
Nit: Since this data is static, you can just declare this as a `var`

jpower432 commented on plugin/transform_test.go:
```diff
@@ -0,0 +1,81 @@
+}
+
+// test harness for a policy request proto message
+func loadPolicyRequestProto() *proto.PolicyRequest {
```
Same

jpower432 reviewed (changes requested):
Looks good. Left a few suggestions.

jpower432 commented on plugin/policy_plugin.go:
```diff
@@ -0,0 +1,50 @@
+}
+
+func (p *pvpService) GetResults(ctx context.Context, request *proto.PolicyRequest) (*proto.ResultsResponse, error) {
+	policy := ProtoToPolicy(request)
```
```suggestion
	policy := NewPolicyFromProto(request)
```

jpower432 committed:
fix: normalize slices in transform.go to nil slices instead of empty

There is a chance many of the slices in the transformation will
not be populated. Using a nil slices does not allocate the underlying
array when it is not needed.

Signed-off-by: Jennifer Power <barnabei.jennifer@gmail.com>

gvauter committed:
Merge pull request #1 from jpower432/CPLYTM-383

fix: normalize slices in transform.go to nil slices instead of empty

jpower432 reviewed (approved):
LGTM
